libc/include/CMakeLists.txt

```diff
@@ -17,1304 +17,697 @@ add_header(
-    DEF_FILE sys/syscall.h.def
-    GEN_HDR sys/syscall.h
-  )
-
-  add_gen_header2(
-    sys_time
-    YAML_FILE ../libc/newhdrgen/yaml/sys/sys_time.yaml
-    DEF_FILE sys/time.h.def
-    GEN_HDR sys/time.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-types.struct_timeval
-      .llvm-libc-macros.sys_time_macros
-  )
-
-  add_gen_header2(
-    sys_types
-    YAML_FILE ../libc/newhdrgen/yaml/sys/sys_types.yaml
-    DEF_FILE sys/types.h.def
-    GEN_HDR sys/types.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-types.blkcnt_t
-      .llvm-libc-types.blksize_t
-      .llvm-libc-types.clockid_t
-      .llvm-libc-types.dev_t
-      .llvm-libc-types.gid_t
-      .llvm-libc-types.ino_t
-      .llvm-libc-types.mode_t
-      .llvm-libc-types.nlink_t
-      .llvm-libc-types.off_t
-      .llvm-libc-types.pid_t
-      .llvm-libc-types.pthread_attr_t
-      .llvm-libc-types.pthread_key_t
-      .llvm-libc-types.pthread_mutex_t
-      .llvm-libc-types.pthread_mutexattr_t
-      .llvm-libc-types.pthread_once_t
-      .llvm-libc-types.pthread_t
-      .llvm-libc-types.size_t
-      .llvm-libc-types.ssize_t
-      .llvm-libc-types.suseconds_t
-      .llvm-libc-types.time_t
-      .llvm-libc-types.uid_t
-  )
-
-  add_gen_header2(
-    sys_utsname
-    YAML_FILE ../libc/newhdrgen/yaml/sys/sys_utsname.yaml
-    DEF_FILE sys/utsname.h.def
-    GEN_HDR sys/utsname.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-types.struct_utsname
-  )
-
-  add_gen_header2(
-    sys_wait
-    YAML_FILE ../libc/newhdrgen/yaml/sys/sys_wait.yaml
-    DEF_FILE sys/wait.h.def
-    GEN_HDR sys/wait.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-macros.sys_wait_macros
-      .llvm-libc-types.pid_t
-      .llvm-libc-types.struct_rusage
-      .llvm-libc-types.siginfo_t
-  )
-
-  add_gen_header2(
-    termios
-    YAML_FILE ../libc/newhdrgen/yaml/termios.yaml
-    DEF_FILE termios.h.def
-    GEN_HDR termios.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-macros.termios_macros
-      .llvm-libc-types.cc_t
-      .llvm-libc-types.pid_t
-      .llvm-libc-types.speed_t
-      .llvm-libc-types.struct_termios
-      .llvm-libc-types.tcflag_t
-  )
-
-  add_gen_header2(
-    uchar
-    YAML_FILE ../libc/newhdrgen/yaml/uchar.yaml
-    DEF_FILE uchar.h.def
-    GEN_HDR uchar.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-types.mbstate_t
-      .llvm-libc-types.char8_t
-      .llvm-libc-types.char16_t
-      .llvm-libc-types.char32_t
-  )
-
-  add_gen_header2(
-    wchar
-    YAML_FILE ../libc/newhdrgen/yaml/wchar.yaml
-    DEF_FILE wchar.h.def
-    GEN_HDR wchar.h
-    DEPENDS
-      .llvm_libc_common_h
-      .llvm-libc-macros.wchar_macros
-      .llvm-libc-types.mbstate_t
-      .llvm-libc-types.size_t
-      .llvm-libc-types.wint_t
-      .llvm-libc-types.wchar_t
-  )
-
-  if(LIBC_TARGET_OS_IS_GPU)
-    file(MAKE_DIRECTORY ${LIBC_INCLUDE_DIR}/gpu)
-
+macro(add_header_macro TYPE YAML_FILE DEF_FILE GEN_HDR DEPENDS)
```

Comment:
I think we can remove type since it is not defined in the libcheader

Reply:
Sorry, that was a misname. TYPE should be called NAME. It should used in both add_gen_header and add_gen_header2 right?